src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  Array.prototype.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    if (separator !== undefined) {
+      separator = String(separator);
```

Comment:
better to create a variable and use it
```suggestion
      const separatorStr = String(separator);
```